terraform/azure/components/nhais/kube-nhais_deployment.tf

```diff
@@ -0,0 +1,163 @@
+
+      spec {
+        container {
+          image = var.nhais_image
+          name = local.resource_prefix
+
+          port {
+            name = "container-port"
+            container_port = "${var.nhais_container_port}"
+            protocol = "TCP"
+          } // port
+
+          env {
+            name = "NHAIS_AMQP_BROKERS"
+            value = "amqps://${replace(replace(split(";", azurerm_servicebus_namespace_authorization_rule.nhais_servicebus_ar.primary_connection_string)[0],"Endpoint=sb://",""),"/","")}:5671/?sasl=plain"
+          }
+
+          env {
+              name = "NHAIS_AMQP_USERNAME"
+              value = azurerm_servicebus_namespace_authorization_rule.nhais_servicebus_ar.name
+          }
+
+          env {
+              name = "NHAIS_AMQP_PASSWORD"
+              value = azurerm_servicebus_namespace_authorization_rule.nhais_servicebus_ar.primary_key
+          }
+
+          env {
+              name = "NHAIS_MESH_OUTBOUND_QUEUE_NAME"
+              value = azurerm_servicebus_queue.nhais_mesh_outbound_queue.name
+          }
+
+          env {
+              name = "NHAIS_MESH_INBOUND_QUEUE_NAME"
+              value = azurerm_servicebus_queue.nhais_mesh_inbound_queue.name
+          }
+
+          env {
+              name = "NHAIS_GP_SYSTEM_INBOUND_QUEUE_NAME"
+              value = azurerm_servicebus_queue.nhais_gp_inbound_queue.name
+          }
+
+          env {
+              name = "NHAIS_MONGO_HOST"
+              value = data.terraform_remote_state.base.outputs.mongodb_hostname
+          }
+
+          env {
+              name = "NHAIS_MONGO_PORT"
+              value = data.terraform_remote_state.base.outputs.mongodb_port
+          }
+
+          env {
+              name = "NHAIS_MONGO_USERNAME"
+              value = data.terraform_remote_state.base.outputs.mongodb_username
+          }
+
+          env {
+              name = "NHAIS_MONGO_PASSWORD"
+              value = data.terraform_remote_state.base.outputs.mongodb_password
+          }
+
+          env {
+              name = "NHAIS_MONGO_OPTIONS"
+              value = "ssl=true&replicaSet=globaldb&retrywrites=false&maxIdleTimeMS=120000&appName=@${data.terraform_remote_state.base.outputs.mongodb_username}"
+          }
+
+          env {
+              name = "NHAIS_COSMOS_DB_ENABLED"
+              value = "true"
+          }
+
+          env {
+              name = "NHAIS_MESH_MAILBOX_ID"
+              value = var.nhais_mesh_mailbox_id
+          }
+
+          env {
+              name = "NHAIS_MESH_MAILBOX_PASSWORD"
+              value = var.nhais_mesh_mailbox_password
+          }
+
+          env {
+              name = "NHAIS_MESH_SHARED_KEY"
+              value = var.nhais_mesh_shared_key
+          }
+
+          env {
+              name = "NHAIS_MESH_HOST"
+              value = var.fake_mesh_in_use ? "https://${local.resource_prefix}-fake-mesh:${var.fake_mesh_application_port}/messageexchange/" : var.nhais_mesh_host
```

Comment:
👍 I was thinking how to do GP2GP mock component - seperate or flagged in/out in the gp2gp component.